app/javascript/components/hocs/with-widget-transition-controls.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import { mathMod } from 'ramda';
+
+const prefetchDelay = 1500;
+
+export const shouldRequestWidget = state =>
+  state.msDurationRemaining < Date.now() - state.startTimestamp;
```

Comment:
Is there a reason we would want to use a 'Date' object over a 'Time' object here?

Reply:
What is a Time object?